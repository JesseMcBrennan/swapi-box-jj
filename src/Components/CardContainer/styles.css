.card-container {
  display: flex;
  overflow: auto;
  width: 100%;
}