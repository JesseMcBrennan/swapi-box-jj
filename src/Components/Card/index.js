import React from 'react';
import PropTypes from 'prop-types';
import './styles.css';

export const Card = ({name, homeworld, species, population }) => {
  return (
    <div>
      <h2>{name}</h2>
      <h2>{homeworld}</h2>
      <h2>{species}</h2>
      <h2>{population}</h2>
    </div>
  );
};

Card.propTypes = {
  name: PropTypes.string,
  homeworld: PropTypes.string,
  species: PropTypes.string,
  population: PropTypes.string
};
