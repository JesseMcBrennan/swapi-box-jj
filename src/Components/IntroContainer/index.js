import React from 'react';
import PropTypes from 'prop-types';
import './styles.css';

export const IntroContainer = ({ filmData }) => {
  const randomNumber = Math.floor(Math.random() * (7));
  const displayFilmData = filmData[`${randomNumber}`];

  return (
    <div className='target'>
      <div className="introContainer">
        <p>{displayFilmData.crawl}</p>
        <h1>{displayFilmData.title}</h1>
        <h2>{displayFilmData.releaseDate}</h2>
      </div>
    </div>
  );
}; 

IntroContainer.propTypes = {
  filmData: PropTypes.object
};