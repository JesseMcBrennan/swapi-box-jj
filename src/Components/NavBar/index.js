import React from 'react'
import './styles.css'

export const NavBar = () => {
  return(
    <div className="navBar">
      <button>People</button>
      <button>Vehicles</button>
      <button>Planets</button>
    </div>
  )
}