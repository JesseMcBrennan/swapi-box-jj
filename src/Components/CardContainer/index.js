import React from 'react';
import PropTypes from 'prop-types';
import { Card } from "../Card";
import './styles.css';

export const CardContainer = ({ peopleData, planetData, vehicleData, favoritesData }) => {
  // const displayPeopleCards = peopleData.map(person => <Card {...person} />);

  return (
    <div>
   {/*   {displayPeopleCards}*/}
    </div>
  );
};

CardContainer.propTypes = {
  peopleData: PropTypes.array
};