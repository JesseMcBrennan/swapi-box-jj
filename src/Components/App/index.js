
import React, { Component } from 'react';
import './styles.css';
import { filmDataCall } from '../../data/apiFetchCall';
import { IntroContainer } from '../IntroContainer';

class App extends Component {
  constructor() {
    super();
    this.state = {
      filmData: []
    };
  }

  componentDidMount() {
    this.retrieveFilmData();
  }

  retrieveFilmData = async () => {
    const filmData = await filmDataCall();
    this.setState({
      filmData
    });
  }

  render() {
    const { filmData } = this.state;
    const displayData = filmData.length ? <IntroContainer filmData={filmData} /> : <h1>place holder</h1>;
    return (
      <div className="App">
        {displayData}
      </div>
    );
  }
}

export default App;
