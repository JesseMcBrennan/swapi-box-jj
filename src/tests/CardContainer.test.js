import React from 'react';
import { shallow } from 'enzyme';
import { CardContainer } from '../Components/CardContainer';

describe('CardContainer', () => {
  let wrapper;
  let mockCardsData;
  let mockToggleFavorites;

  beforeEach(() => {
    mockCardsData = [{cardInfo: 'card value'}, {cardInfo: 'card value'}];
    mockToggleFavorites = jest.fn();
    wrapper = shallow(
      <CardContainer 
        cardsData={mockCardsData} 
        toggleFavorites={mockToggleFavorites}
      />);
  });

  it('renders the CardContainer with the correct props', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
