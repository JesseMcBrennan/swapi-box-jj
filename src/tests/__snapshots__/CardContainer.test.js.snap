// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer renders the CardContainer with the correct props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    cardsData={
        Array [
            Object {
              "cardInfo": "card value",
            },
            Object {
              "cardInfo": "card value",
            },
          ]
    }
    toggleFavorites={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          data={
                    Object {
                              "cardInfo": "card value",
                            }
          }
          toggleFavorites={[Function]}
/>,
        <Unknown
          data={
                    Object {
                              "cardInfo": "card value",
                            }
          }
          toggleFavorites={[Function]}
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "data": Object {
            "cardInfo": "card value",
          },
          "toggleFavorites": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "data": Object {
            "cardInfo": "card value",
          },
          "toggleFavorites": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            data={
                        Object {
                                    "cardInfo": "card value",
                                  }
            }
            toggleFavorites={[Function]}
/>,
          <Unknown
            data={
                        Object {
                                    "cardInfo": "card value",
                                  }
            }
            toggleFavorites={[Function]}
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "cardInfo": "card value",
            },
            "toggleFavorites": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "cardInfo": "card value",
            },
            "toggleFavorites": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
